pkg/packetbrokeragent/translation.go

```diff
@@ -217,13 +256,28 @@ func unwrapGatewayUplinkToken(token, key []byte) (string, []byte, error) {
 	if err != nil {
 		return "", nil, err
 	}
-	var t gatewayUplinkToken
+	var t legacyGatewayUplinkToken
 	if err := json.Unmarshal(plaintext, &t); err != nil {
 		return "", nil, err
 	}
 	return t.GatewayUID, t.Token, nil
 }
 
+func unwrapGatewayUplinkToken(token []byte, aead cipher.AEAD, legacyKey []byte) (string, []byte, error) {
+	if uid, token, err := unwrapLegacyGatewayUplinkToken(token, legacyKey); err == nil {
+		return uid, token, nil
+	}
+	plaintext, err := decryptCiphertext(token, aead)
+	if err != nil {
+		return "", nil, err
+	}
```

Comment:
Please file issue and reference here to remove the legacy handling here

Reply:
I'll create `v3.29` already and open a PR removing the legacy contents towards that.